### ISSUE TITLE
Add error code to the default virtualtext message.

### ISSUE THREAD

**Issue by w0rp** [dense-analysis/ale#4412]
Often for linters such as shellcheck the information you immediately need to see is an error code so you can Google the error code and read the wiki for how to deal with the problem, or to ignore a problem where you know it's fine to ignore. To that and, we can include short errors codes in the default ALE virtualtext message, which people can configure how they like. The default experience should be immediately revealing problems.

**Comment by w0rp** [comment]
I think to do this we should support a variation of error codes where we only display error codes that are actually short codes. Nobody wants to see this:

```
foo() # E(some-very-long-error-name): Something is wrong here
```

People will like this:

```
foo() # E(SC1234): Fix this please
```

We can probably just do this with a simple string length check.

Bearing in mind in the absence of a code, ALE can easily display this instead:

```
foo() # E: Fix this please
```

**Comment by RyanSquared** [comment]
It would be nice to have the length cutoff to be a configurable option. For something like Rust, knowing which lint to disable inline in the code would be handy. For example:

```rust
return x; // E(needless_return): unneeded `return` statement
```

and now I can make that:

```rust
#[allow(needless_return)]
return x;
```

**Comment by w0rp** [comment]
> It would be nice to have the length cutoff to be a configurable option. For something like Rust, knowing which lint to disable inline in the code would be handy. For example:
> 
> ```rust
> return x; // E(needless_return): unneeded `return` statement
> ```
> 
> and now I can make that:
> 
> ```rust
> #[allow(needless_return)]
> return x;
> ```

Yes, that's what I'm thinking of. It can be just a number, and it can also apply to `ale_echo_msg_format`. The same function is called to format messages for pretty much everything, so all things benefit from the same patches.

**Comment by ououmania** [comment]
Excuse me, how's this going? I am looking for this feature exactly.
